cmd/app.go

```diff
@@ -0,0 +1,25 @@
+// Copyright 2024
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"github.com/Mirantis/hmc/pkg/manager"
+	_ "github.com/Mirantis/hmc/pkg/providers/azure"
```

Comment:
No, due to how unit-tests are structured `aws` is enabled unconditionally.

Basically, unit-tests (mostly webhooks) use `aws` fixtures for testing, so there is no reason to have that provider optionally disable and will also break tests.

To make this possible some `dummy` provider is needed, or we can have `aws` and avoid questions why `dummy` provider exists

Reply:
Not from tests, due to tests `aws` is registered [in](https://github.com/s3rj1k/hmc/blob/provider-modules/pkg/providers/providers.go#L150), which makes `aws` a defiantly Tier-1 provider.

I've gently removed `azure` unit-tests dependency, so it's only `aws` is like this, forever loved and supported provider.

[This](https://github.com/s3rj1k/hmc/blob/provider-modules/pkg/providers/aws/provider.go#L26) basically tells this story, not sure that whole unit-test story needs to be put in comment, if you think this will benefit us, please suggest what to put there.

Reply:
yep, due to the import from the tests. that's clear. ok then, I think we can skip this one, I will leave this comment for a while if any of the rest reviewers would like to participate, otherwise, I'll resolve it